### PR TITLE
feat(docs/internal): :sparkles: Add documentation/guiedlines on writing good error/exception messages in our libraries.

### PR DESCRIPTION
**Pull request by pklaschka**
Adds guidelines for writing error/exception messages for our library components. This is crucial for a good developer experience and should get used by anyone contributing to any of our library repos.

Yes, I have signed the WüSpace CLA.

### CODE REVIEW

File: docs/internal/concepts/error-messages.mdx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+Error messages are a part of practically every application or library we
+develop. Especially in libraries (like `telestion-core` or `telestion-client`),
+good error messages are crucial for a good developer experience.
+
+:::warning
+
+Developers using your libraries **shouldn't have to know** how your library
+works to solve problems. It is therefore crucial to handle potential errors of a
+library in the library itself and, if necessary, throw (to the developer using
+the library) only higher-level exceptions with good error messages.
```
Comment (cb0s) [inline docs/internal/concepts/error-messages.mdx:54]:
```suggestion
library in the library itself and, if necessary, throw only higher-level exceptions with good error messages to the developer using the library.
```

File: docs/internal/concepts/error-messages.mdx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+However well-written an error message might be: A few abstractions down the line
```
Comment (cb0s) [inline docs/internal/concepts/error-messages.mdx:56]: Hm... Not the best formulation... I kind of struggled the first time reading...

File: docs/internal/concepts/error-messages.mdx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+and provide **their own** error messages that are helpful in **the context of the
+immediate caller** of that function to guarantee for a good developer experience!_
+
+:::
+
+## Error messages for expected errors
+
+It might seem strange, but most errors are actually "expected" errors. For
+example, when opening a file, we can already anticipate that the user might not
+have the rights to read the file and the operation might, therefore, fail.
```
Comment (cb0s) [inline docs/internal/concepts/error-messages.mdx:70]:
```suggestion
have the rights to read the file and the operation might therefore fail.
```

File: docs/internal/references/actionable-error-message-structure.mdx [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+:::
+
+## AWESA: Actionable Error Message components
+
+Actionable Error Messages should follow the **AWESA** (inspired by the Swahili
+term "kuweza" = "to be able to") structure:
+
+### Attention
+
+Describe, as quickly as possible, that an error occurred and what type of error
```
Comment (cb0s) [inline docs/internal/references/actionable-error-message-structure.mdx:66]:
```suggestion
Describe as quickly as possible, that an error occurred and what type of error
```

**Review by cb0s — CHANGES_REQUESTED** [review]
I just overlooked the PR: In general I quite like, but there some minor things I would like to see fixed.

File: docs/internal/references/actionable-error-message-structure.mdx (hunk 4 above)
Comment (pklaschka) [reply docs/internal/references/actionable-error-message-structure.mdx:66]: Nope, sorry.

**Review by fussel178 — DISMISSED** [review]
LGTM. Happy christmas :christmas_tree:

File: docs/internal/references/unexpected-error-message-structure.mdx [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+## NADAD: Unepxected Error Message components
+
+Unexpected error messages should follow the **NADAD** structure:
+
+### Notify
+
+Developers appreciate directly getting information, even if it's information
+that they don't like. Therefore, you should notify them what that and where an
+unexpected error occurred:
```
Comment (cb0s) [inline docs/internal/references/unexpected-error-message-structure.mdx:60]:
```suggestion
that they don't like. Therefore, you should notify them what caused that unexpected error and where an
it occurred:
```

**Review by cb0s — CHANGES_REQUESTED** [review]
One last minor thing I would write a little differently before in my opinion it is ready to be merged

**Review by cb0s — APPROVED** [review]
✔️

**Review by fussel178 — APPROVED** [review]
:santa:

**Pull request merged**